Shared/Supporting Files/Views/CategoriesView.swift

```diff
@@ -15,39 +15,70 @@
+        var categories = ["All", "Favorites"]
+        let sampleCategories = Set(SamplesApp.samples.map(\.category))
+        categories.append(contentsOf: sampleCategories.sorted())
+        return categories
+    }()
     
     var body: some View {
         ScrollView {
             LazyVGrid(columns: [GridItem(), GridItem()]) {
-                NavigationLink {
-                    FavoritesView()
-                        .navigationTitle("Favorites")
-                } label: {
-                    CategoryTile(name: "Favorites")
-                }
-                .isDetailLink(false)
-                .buttonStyle(.plain)
-                .contentShape(RoundedRectangle(cornerRadius: 30))
-                
                 ForEach(categories, id: \.self) { category in
-                    NavigationLink {
-                        List(SamplesApp.samples.filter { $0.category == category }, id: \.name) { sample in
-                            SampleLink(sample)
-                        }
-                        .listStyle(.sidebar)
-                        .navigationTitle(category)
+                    Button {
+                        selectedCategory = category
+                        columnVisibility = .doubleColumn
                     } label: {
                         CategoryTile(name: category)
                     }
-                    .isDetailLink(false)
                     .buttonStyle(.plain)
-                    .contentShape(RoundedRectangle(cornerRadius: 30))
+                    .contentShape(RoundedRectangle(cornerRadius: 15))
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 15)
+                            .stroke(
+                                category == selectedCategory ? Color.accentColor : Color.clear,
+                                lineWidth: 5
+                            )
+                    )
                 }
             }
             .padding()
         }
+        .navigationDestination(isPresented: $destinationIsPresented) {
+            Group {
+                switch selectedCategory {
+                case "All":
+                    AllSamplesView()
+                case "Favorites":
+                    FavoritesView()
+                default:
+                    List(SamplesApp.samples.filter { $0.category == selectedCategory }, id: \.name) {
+                        SampleLink($0)
+                    }
+                }
+            }
+            .listStyle(.insetGrouped)
+            .navigationTitle(selectedCategory ?? "")
+        }
+        .onChange(of: destinationIsPresented) { _ in
+            // Resets the selection when the navigation destination is no longer presented.
+            guard !destinationIsPresented else { return }
+            selectedCategory = nil
+        }
+        .onChange(of: selectedCategory) { newSelection in
+            destinationIsPresented = newSelection != nil
+        }
     }
 }
 
```

Comment:
While no change happened to the tile in this PR, I'd suggest setting the overlay's size relative to the background Image, so the icon doesn't take too much space that causes the text to be clipped.

![clipped](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/9660181/ed0a1b3a-0217-4c64-986c-e4172d4cee94)

Since it is not related to the split view changes, ok to do it in a separate PR if you want.

Reply:
Issue: `swift/issues/5250`